packages/alea-frontend/pages/blog/[postId].tsx

```diff
@@ -116,16 +136,40 @@ const BlogPostPage: NextPage = ({ post }: { post: BlogPost }) => {
 
 export default BlogPostPage;
 
-export async function getServerSideProps(context: GetServerSidePropsContext) {
-  const postId = context.query.postId as string;
-  const protocol = context.req.headers['x-forwarded-proto'] as string;
-  const host = context.req.headers.host;
+export async function getStaticPaths() {
+  const data = fs.readFileSync('../../static/blogData.json', 'utf-8');
+  const jsonData = JSON.parse(data);
+  const paths = jsonData.map((blog: BlogPost) => ({
+    params: { postId: blog.postId },
+  }));
+  return { paths, fallback: true };
+}
+
+export async function getStaticProps({
+  params,
+}: {
+  params: { postId: string };
+}) {
   try {
-    const post = await getPostById(postId, true, protocol, host);
-    console.log(post);
-    return { props: { post } };
+    const data = fs.readFileSync('../../static/blogData.json', 'utf-8');
+    const jsonData = JSON.parse(data);
+    const post = jsonData.find(
+      (blog: BlogPost) => blog.postId === params.postId
+    );
+    if (!post) {
+      return { props: { post: null } };
+    }
+    return {
+      props: {
+        post,
+      },
+    };
   } catch (error) {
-    console.error('Error fetching data:', error);
-    return { props: { data: null } };
+    console.error('Error fetching blog post:', error);
+    return {
```

Comment:
We need to talk about revalidation and the need to move `blogData.json` to the compiled project `dist`.